### PR TITLE
Add uptime to crawl data (RIPD-997)

### PR DESCRIPTION
**Pull request by miguelportilla**
Reviewers: @HowardHinnant @scottschurr

### CODE REVIEW

**Comment by rec** [comment]
Scott has more reviews outstanding than anyone else - I could take it over?

**Comment by scottschurr** [comment]
@rec, that would be fine with me.

**Comment by HowardHinnant** [comment]
:+1:

**Comment by scottschurr** [comment]
LGTM :+1:

File: src/ripple/overlay/impl/OverlayImpl.cpp [hunk 1]
```diff
@@ -752,12 +753,14 @@ OverlayImpl::crawl()
...
         {
             auto& pv = av.append(Json::Value(Json::objectValue));
-            pv[jss::type] = "peer";
             pv[jss::public_key] = beast::base64_encode(
                 sp->getNodePublic().getNodePublic().data(),
                     sp->getNodePublic().getNodePublic().size());
             pv[jss::type] = sp->slot()->inbound() ?
                 "in" : "out";
+            pv[jss::uptime] =
+                static_cast<std::uint32_t>(duration_cast<seconds>(
```
Comment (rec) [inline src/ripple/overlay/impl/OverlayImpl.cpp]: Does it not work without that outermost cast?

**Comment by rec** [comment]
:+1:

File: src/ripple/overlay/impl/OverlayImpl.cpp (hunk 1 above)
Comment (miguelportilla) [inline src/ripple/overlay/impl/OverlayImpl.cpp]: Unfortunately not.

**Pull request closed without merging**